incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -300,12 +348,14 @@ func (r *HierarchyConfigReconciler) syncAnchors(log logr.Logger, ns *forest.Name
 }
 
 func (r *HierarchyConfigReconciler) syncLabel(log logr.Logger, nsInst *corev1.Namespace, ns *forest.Namespace) {
-	// Pre-define label depth suffix
-	labelDepthSuffix := fmt.Sprintf(".tree.%s/depth", api.MetaGroup)
+	if ns.IsExternal() {
+		metadata.SetLabel(nsInst, nsInst.Name+api.LabelTreeDepthSuffix, "0")
+		return
+	}
```

Comment:
Updated here to always set tree label of itself on external namespaces and updated the test cases. (Previously, HNC didn't change/set any labels on external namespaces and only automatically add the tree label of itself with `depth=0` in the forest).

Reply:
Can you add this to the design doc?